src/Akka.Persistence.MongoDb/Snapshot/MongoDbGridFSSnapshotStore.cs

```diff
@@ -0,0 +1,356 @@
+        if (!setupOption.HasValue || setupOption.Value.SnapshotConnectionSettings == null)
+        {
+            //Default LinqProvider has been changed to LINQ3.LinqProvider can be changed back to LINQ2 in the following way:
+            var connectionString = new MongoUrl(_settings.ConnectionString);
+            var clientSettings = MongoClientSettings.FromUrl(connectionString);
+            clientSettings.LinqProvider = LinqProvider.V2;
+            client = new MongoClient(clientSettings);
+            _mongoDatabase_DoNotUseDirectly = client.GetDatabase(connectionString.DatabaseName);
+            return _mongoDatabase_DoNotUseDirectly;
+        }
+
+        client = new MongoClient(setupOption.Value.SnapshotConnectionSettings);
+        _mongoDatabase_DoNotUseDirectly = client.GetDatabase(setupOption.Value.SnapshotDatabaseName);
+        return _mongoDatabase_DoNotUseDirectly;
+    }
+
+    private GridFSBucket GetGridFSBucket()
+    {
+        var db = GetMongoDb();
+        _snapshotGridFSBucket_DoNotUseDirectly = new GridFSBucket(db, _bucketOptions);
+
+        return _snapshotGridFSBucket_DoNotUseDirectly;
+    }
+
+    private IMongoCollection<GridFSFileInfo> GetFilesCollection()
+    {
+        return GetMongoDb().GetCollection<GridFSFileInfo>(_settings.Collection + ".files");
+    }
+
+    private IMongoCollection<BsonDocument> GetChunksCollection()
+    {
+        return GetMongoDb().GetCollection<BsonDocument>(_settings.Collection + ".chunks");
+    }
+    
+    protected override void PostStop()
+    {
+        // cancel any pending database commands during shutdown
+        _pendingCommandsCancellation.Cancel();
+        _pendingCommandsCancellation.Dispose();
+        base.PostStop();
+    }
+
+    protected override async Task<SelectedSnapshot?> LoadAsync(string persistenceId, SnapshotSelectionCriteria criteria)
+    {
+        using var unitedCts = CreatePerCallCts();
+        var filesCollection = GetFilesCollection();
+        var chunksCollection = GetChunksCollection();
+        
+        var filter = CreateRangeFilter(persistenceId, criteria);
+
+        return await MaybeWithTransaction<SelectedSnapshot?>(async (session, token) =>
+        {
+            var info = await (session is not null 
+                    ? filesCollection.Find(session, filter) 
+                    : filesCollection.Find(filter)) 
+                .SortByDescending(x => x.Metadata[SequenceNrKey])
+                .Limit(1)
+                .FirstOrDefaultAsync(token);
+            
+            if (info is null)
+                return null;
+            
+            var chunkFilter = Builders<BsonDocument>.Filter.Eq(doc => doc["files_id"], info.Id);
+            var chunkCursor = await (session is not null
+                    ? chunksCollection.Find(session, chunkFilter)
+                    : chunksCollection.Find(chunkFilter))
+                .SortBy(x => x["n"])
+                .ToCursorAsync(token);
```

Comment:
Reason for the slower load times for larger snapshot sizes is that we're loading snapshot chunks from GridFS sequentially - we could potentially do this concurrently